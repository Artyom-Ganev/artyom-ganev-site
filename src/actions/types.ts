import { BlogModel, CareerModel } from 'model';
import { AnyAction } from 'redux';

export interface IAppLoadingAction extends AnyAction {
  loading: boolean;
}

export interface IErrorAction extends AnyAction {
  error: string;
}

export interface ILoadBlogListSuccessAction extends AnyAction {
  items: BlogModel[];
}

export interface ILoadCareerListSuccessAction extends AnyAction {
  items: CareerModel[];
}

export type AppActionTypes = IAppLoadingAction | ILoadBlogListSuccessAction | ILoadCareerListSuccessAction | AnyAction;

export type BlogActionsType = ILoadBlogListSuccessAction | IErrorAction | AnyAction;

export type CareerActionsType = ILoadCareerListSuccessAction | IErrorAction | AnyAction;
