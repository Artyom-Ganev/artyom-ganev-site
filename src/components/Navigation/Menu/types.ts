import { IPageProps } from 'components/Pages';
import { IClickable } from 'types';

export interface INavigationMenuItem extends IPageProps, IClickable {}
