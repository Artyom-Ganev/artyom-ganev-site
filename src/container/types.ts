import { OverridableComponent } from '@material-ui/core/OverridableComponent';
import { SvgIconTypeMap } from '@material-ui/core/SvgIcon/SvgIcon';

export interface IPageProps {
  id: string;
  Icon: OverridableComponent<SvgIconTypeMap>;
  name: string;
  url: string;
}
