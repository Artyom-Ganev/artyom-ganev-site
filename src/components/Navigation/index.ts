export { default as LoadingIndicator } from './LoadingIndicator';
export { default as NavigationMenu } from './Menu';
export { default as Routes } from './Routes';
