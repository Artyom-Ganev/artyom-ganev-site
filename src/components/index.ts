export { default as App, NarrowContext } from './App';
