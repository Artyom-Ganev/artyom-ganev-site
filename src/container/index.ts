export { default as PageContainer } from './Page';
