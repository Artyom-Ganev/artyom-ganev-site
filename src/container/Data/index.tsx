import React, { Component, ReactNode } from 'react';
import { PageContainer } from 'src/container';
import { IDataProps, Source } from 'src/interface';

/**
 * Abstract page container with data source
 */
abstract class AbstractDataContainer<TData, TProps extends IDataProps<TData> = IDataProps<TData>> extends Component<
  TProps
> {
  public componentDidMount(): void {
    const { items } = this.props;
    if (!items.length) {
      this.getSource().loadList();
    }
  }

  public render(): ReactNode {
    const { items } = this.props;
    return <PageContainer>{this.getContent(items)}</PageContainer>;
  }

  protected abstract getSource(): Source;

  protected abstract getContent(items: TData[]): ReactNode | ReactNode[];
}

export default AbstractDataContainer;
