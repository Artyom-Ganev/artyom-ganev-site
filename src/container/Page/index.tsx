import React, { FunctionComponent, PropsWithChildren, useContext } from 'react';
import { useSelector } from 'react-redux';
import { NarrowContext } from 'src/components';
import { LoadingIndicator } from 'src/components/Navigation';
import { isLoading } from 'src/selectors';
import { IState } from 'src/types';
import styles from './Container.module.scss';

/**
 * Container properties interface
 */
interface IProperties {
  title?: string;
}

/**
 * Page container with title
 */
const Container: FunctionComponent<PropsWithChildren<IProperties>> = ({
  children,
  title,
}: PropsWithChildren<IProperties>) => {
  const loading = useSelector<IState, boolean>(isLoading);
  const narrow = useContext(NarrowContext);
  const contentClass = `flexBox flexColumn ${styles.container} ${narrow ? styles.narrow : styles.wide}`;
  return loading ? (
    <div className={contentClass}>
      <LoadingIndicator />
    </div>
  ) : (
    <div className={contentClass}>
      {title && <h2>{title}</h2>}
      {children}
    </div>
  );
};

export default Container;
