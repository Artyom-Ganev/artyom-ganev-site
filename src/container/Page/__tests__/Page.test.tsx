import { mount, ReactWrapper } from 'enzyme';
import toJson from 'enzyme-to-json';
import React, { FunctionComponent } from 'react';
import { Provider } from 'react-redux';
import { combineReducers, createStore } from 'redux';
import { LoadingIndicator } from 'src/components/Navigation';
import { PageContainer } from 'src/container';
import { IAppState } from 'src/types';

describe('Container', () => {
  const titleMock = 'title 1';
  const Content: FunctionComponent = () => <div>text1</div>;

  const render = (loading = false): ReactWrapper => {
    const app = (): IAppState => ({ loading });
    const store = createStore(combineReducers({ app }));
    return mount(
      <Provider store={store}>
        <PageContainer title={titleMock}>
          <Content />
        </PageContainer>
      </Provider>
    );
  };

  describe('loading', () => {
    const component = render(true);

    it('Should render loader', () => {
      expect(component.find(LoadingIndicator)).toHaveLength(1);
    });

    it("Shouldn't render title", () => {
      const title = component.find('h2');
      expect(title).toHaveLength(0);
    });

    it("Shouldn't render content", () => {
      expect(component.find(Content)).toHaveLength(0);
    });
  });

  describe('loaded', () => {
    const component = render();

    it("Shouldn't render loader", () => {
      expect(component.find(LoadingIndicator)).toHaveLength(0);
    });

    it('Should render title', () => {
      const title = component.find('h2');
      expect(title).toHaveLength(1);
      expect(toJson(title)).toMatchSnapshot();
    });

    it('Should render content', () => {
      expect(component.find(Content)).toHaveLength(1);
    });
  });
});
