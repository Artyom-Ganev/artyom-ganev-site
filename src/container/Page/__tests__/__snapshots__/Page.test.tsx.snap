// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container loaded Should render title 1`] = `
<h2>
  title 1
</h2>
`;
